extensions/amp-story/1.0/amp-story.js

```diff
@@ -2237,13 +2240,34 @@ export class AmpStory extends AMP.BaseElement {
 
     const pagesByDistance = this.getPagesByDistance_();
 
-    this.mutateElement(() => {
+    const preloadAllPages = () => {
       pagesByDistance.forEach((pageIds, distance) => {
         pageIds.forEach((pageId) => {
           const page = this.getPageById(pageId);
           page.setDistance(distance);
         });
       });
+    };
+
+    this.mutateElement(() => {
+      if (
+        !isExperimentOn(this.win, 'amp-story-load-first-page-only') ||
+        !prioritizeActivePage
+      ) {
+        return preloadAllPages();
+      }
+
+      const activePageId = devAssert(pagesByDistance[0][0]);
+      new Promise((res, rej) => {
```

Comment:
Did you intend to return this promise? I see some branches of this function return values, others do not, and the JSDoc doesn't have a `@return` tag

Reply:
These returns are just used for flow control (short circuiting the logic), so no need to return the promise as we don't have a use for it